libraries/AP_Math/polyfit.cpp

```diff
@@ -0,0 +1,46 @@
+/*
+  polynomial fitting class, originally written by Siddharth Bharat Purohit
+  re-worked for ArduPilot by Andrew Tridgell
+*/
+
```

Comment:
Should we add the explanation of math used here?

Reply:
sounds good, do you want to do a patch?

Reply:
Sure, can do.